libs/core/src/utils/overlay.ts

```diff
@@ -0,0 +1,171 @@
+export const positionTooltip = (elem, elemPlacement, overlay, offset = 8) => {
+  if (elem == undefined) return;
+
+  const rect = elem.getBoundingClientRect();
+  const contentRect = overlay.getBoundingClientRect();
+
+  // Exit the function if the placement is not set
+  if (!elemPlacement || elemPlacement == "") return;
+
+  if (elemPlacement.includes("right")) {
+    overlay.style.top = '50%';
+    overlay.style.left = `calc(${rect.width}px + ${offset}px)`;
+    overlay.style.transform = 'translateY(-50%)';
+
+    if (elemPlacement.includes("start")) {
+      overlay.style.top = '0';
+      overlay.style.transform = 'translateY(0)';
+    }
+
+    if (elemPlacement.includes("end")) {
+      overlay.style.bottom = '0';
+      overlay.style.top = 'initial';
+      overlay.style.transform = 'translateY(0)';
+    }
+  }
```

Comment:
Is there a time it can be `right` and `left`? Probably not. Lets convert this to a `Switch` statement with the additional `if` statements in each section

```
switch (true) {

    case elemPlacement.includes("right"): 
        overlay.style.top = '50%';
        overlay.style.left = `calc(${rect.width}px + ${offset}px)`;
        overlay.style.transform = 'translateY(-50%)';

        if (elemPlacement.includes("start")) {
            overlay.style.top = '0';
            overlay.style.transform = 'translateY(0)';
        }

        if (elemPlacement.includes("end")) {
            overlay.style.bottom = '0';
            overlay.style.top = 'initial';
            overlay.style.transform = 'translateY(0)';
         }

        break;

     case elemPlacement.includes("left"): ...
}
```